sogehige/sogeBot
# feat(permissions): refactor permissions to be dynamic

###### NOTE
All user defined permissions must be updated again, no migration

###### TODO
- [x] update UI permission set for commands
- [x] add command !permission list
- [x] test permission checks
- [x] update custom vars permissions
  - [x] tests update
- update alias permission
- update custom commands permissions
- [x] update system/games core default permissions in options
- [ ] ~migration from legacy permissions to dynamic permissions~ Add release note warning about resetting of permissions
- [x] add waterfall check toggle (only this permission group is allowed and not higher) or add toggle to set allowed permissions
- [x] add `changes pending` warning on permission update
- [x] order should be updated on edit component when changed on list component

###### CHECKLIST



- [x] locales are changed or added
- [x] commit message follows [commit guidelines](https://github.com/sogehige/sogeBot/blob/master/CONTRIBUTING.md#commit-guidelines)
